src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerValidator.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.service;
+
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.dto.GeneralPartnerDataDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.dto.GeneralPartnerDto;
+
+@Component
+public class GeneralPartnerValidator {
+
+    public void isValid(GeneralPartnerDto generalPartnerDto) throws NoSuchMethodException, MethodArgumentNotValidException {
+        var methodParameter = new MethodParameter(GeneralPartnerDataDto.class.getConstructor(), -1);
+        BindingResult bindingResult = new BeanPropertyBindingResult(generalPartnerDto, GeneralPartnerDataDto.class.getName());
+
+        var generalPartnerDataDto = generalPartnerDto.getData();
+
+        if (generalPartnerDataDto.getLegalEntityRegisterName() != null || generalPartnerDataDto.getLegalForm() != null) {
+            checkNotNullLegalEntity(generalPartnerDataDto, bindingResult);
+        } else if (generalPartnerDataDto.getForename() != null || generalPartnerDataDto.getSurname() != null) {
+            checkNotNullPerson(generalPartnerDataDto, bindingResult);
+            isSecondNationalityDifferent(generalPartnerDto, bindingResult);
+        } else {
+            var fieldError = new FieldError(GeneralPartnerDataDto.class.getName(), "", "Some fields are missing");
+            bindingResult.addError(fieldError);
+        }
+
+        if (bindingResult.hasErrors()) {
+            throw new MethodArgumentNotValidException(methodParameter, bindingResult);
+        }
+    }
+
+    private void checkNotNullLegalEntity(GeneralPartnerDataDto generalPartnerDataDto,
+                                         BindingResult bindingResult) {
+        if (generalPartnerDataDto.getLegalEntityRegisterName() == null) {
+            var fieldError = new FieldError(GeneralPartnerDataDto.class.getName(), "legal_entity_register_name", "Legal Entity Register Name is required");
+            bindingResult.addError(fieldError);
+        }
+
+        if (generalPartnerDataDto.getLegalForm() == null) {
```

Comment:
for the null field error checks, you could have a private method do the check and add the error to the bindingResult. Pass in the field value, name, errorMessage and bindingResult